src/core/lp_lexer.ml

```diff
@@ -0,0 +1,216 @@
+(** Lexer for Lambdapi syntax. Uses Sedlex, a Utf8 friendly lexer. Some helper
+    functions to check if a string conflicts with the syntax are also
+    provided. *)
+open Sedlexing
+open Pos
+
+type token =
+  | EOF
+  | L_PAREN | R_PAREN | L_SQ_BRACKET | R_SQ_BRACKET
+  | L_CU_BRACKET | R_CU_BRACKET
+  | APPLY | ARROW | AS | ASSERT | ASSERT_NOT | ASSIGN | ASSOC of Syntax.assoc
+  | AT
+  | BUILTIN | BACKQUOTE
+  | COLON | COMMA | COMPUTE | COMPUTE_TYPE | CONSTANT
+  | DEBUG_FLAGS of (bool * string)
```

Comment:
Are parentheses really needed? Idem for ID.

Reply:
I think they are, we need to represent the identifiers and parentheses in the syntax, don't we?

Reply:
Ah I remember now, they are needed by menhir, who takes constructor with *one* argument, which may be a tuple

Reply:
I see. Thanks to add this in a comment then to remember it.